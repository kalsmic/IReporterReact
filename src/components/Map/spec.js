import React from 'react';
import { shallow } from 'enzyme';
import { Map, Marker } from 'react-leaflet';
import IncidentMap from './index';
import ViewIncidentMap from './viewIncident';


describe('Incident Map', () => {
  const props = {
    hasLocation: true,
    handleMapClick: jest.fn(),
    latlng: {
      lat: 12,
      lng: 120
    },
    mapReference: {

    },
    mapClassName: 'mapClassName',
  };

  let wrapper = shallow(<IncidentMap {...props} />);

  it('should render without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a map', () => {
    expect(wrapper.find(Map).exists());
  });

  it('should show a marker if location is present', () => {
    expect(wrapper.find(Marker).exists());
    expect(wrapper.find(Marker).text()).toContain(12);
    expect(wrapper.find(Marker).text()).toContain(120);

    props.hasLocation = false;

    wrapper = shallow(<IncidentMap {...props} />);
    expect(wrapper.find(Marker).exists()).toBe(false);
  });
});


describe('View Incident Map', () => {
  const props = {
    latlng: {
      lat: 23,
      lng: 24
    },
    mapClassName: 'mapClassName',
  };

  const wrapper = shallow(<ViewIncidentMap {...props} />);

  it('should render without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a map', () => {
    expect(wrapper.find(Map).exists());
  });

  it('should show a marker if locatin is present', () => {
    expect(wrapper.find(Marker).exists());
    expect(wrapper.find(Marker).text()).toContain(23);
    expect(wrapper.find(Marker).text()).toContain(24);
  });
});
