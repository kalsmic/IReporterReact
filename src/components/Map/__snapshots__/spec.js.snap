// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Incident Map should render without crashing 1`] = `
<Map
  center={
    Object {
      "lat": 12,
      "lng": 120,
    }
  }
  className="mapClassName"
  length={4}
  onClick={[MockFunction]}
  onLocationfound={[MockFunction]}
  zoom={13}
>
  <ForwardRef(Leaflet(TileLayer))
    attribution="&copy <a href=\\"http://osm.org/copyright\\">OpenStreetMap</a> contributors"
    url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
  />
  <ForwardRef(Leaflet(Marker))
    draggable={true}
    position={
      Object {
        "lat": 12,
        "lng": 120,
      }
    }
  >
    <ForwardRef(Leaflet(Popup))>
       
      Current location:
      <pre>
        {
  "lat": 12,
  "lng": 120
}
      </pre>
    </ForwardRef(Leaflet(Popup))>
  </ForwardRef(Leaflet(Marker))>
</Map>
`;

exports[`View Incident Map should render without crashing 1`] = `
<Map
  center={
    Object {
      "lat": 23,
      "lng": 24,
    }
  }
  className="mapClassName"
  length={4}
  zoom={13}
>
  <ForwardRef(Leaflet(TileLayer))
    attribution="&copy <a href=\\"http://osm.org/copyright\\">OpenStreetMap</a> contributors"
    url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
  />
  <ForwardRef(Leaflet(Marker))
    draggable={true}
    position={
      Object {
        "lat": 23,
        "lng": 24,
      }
    }
  >
    <ForwardRef(Leaflet(Popup))>
       
      Current location:
      <pre>
        {
  "lat": 23,
  "lng": 24
}
      </pre>
    </ForwardRef(Leaflet(Popup))>
  </ForwardRef(Leaflet(Marker))>
</Map>
`;
